public/lib/helpers/timeline_helper.js

```diff
@@ -29,6 +31,60 @@ define(function (require) {
       }
     };
 
+    TimelineHelper.prototype.pluckLabel = function (hit, params, notify) {
```

Comment:
In my comments above, I thought `pluckLabel` was only doing the following:

```js
let field;
if (params.labelFieldSequence) { // in kibi, we have the path property of a field
  field = kibiUtils.getValuesAtPath(hit._source, params.labelFieldSequence);
} else {
  field = _.get(hit._source, params.labelField);
}

if (field) {
  if (this.isMultivalued(field)) {
    notify.warning('Label field [' + params.labelField + '] is multivalued - the first value will be used.');
  }
  label = this.pickFirstIfMultivalued(field, '');
}
```

I think you could break this method in two:

- one with the above code; and
- the other with the code specific to the highlighting.